### PR TITLE
Update servers when newer version is available.

### PR DESCRIPTION
**Pull request by jvanbaarsen**
* Show a warning banner if the Dokku version is out of date
* Allow servers with dokku > 0.4.11 to auto update
* Servers with dokku < 0.4.11 need to manually update, show instructions

Fixes: #45

### CODE REVIEW

File: app/controllers/servers_controller.rb [hunk 1]
```diff
@@ -61,6 +61,28 @@ def start_installation
     redirect_to server_path(@server)
   end
 
+  def start_update
+    @server = Server.find(params[:id])
+    if VersionInfo.parse(@server.dokku_version) < VersionInfo.parse("v0.4.11")
```
Comment (joshuajansen) [inline app/controllers/servers_controller.rb]: Shouldn't we move this hard-coded reference to `v.0.4.11` to something more abstract/dynamic?

File: app/models/server.rb [hunk 2]
```diff
@@ -24,6 +24,14 @@ def service_status(service)
     end
   end
 
+  def up_to_date?
+    VersionInfo.parse(dokku_version) == VersionInfo.parse(latest_dokku_version)
+  end
+
+  def latest_dokku_version
+    "v0.4.11".freeze
```
Comment (joshuajansen) [inline app/models/server.rb]: Ah, there it is :wink:

File: test/jobs/update_server_job_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UpdateServerJobTest < ActiveJob::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (joshuajansen) [inline test/jobs/update_server_job_test.rb]: Let's add a test or remove this file :smile:

File: lib/version_info.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class VersionInfo
```
Comment (joshuajansen) [inline lib/version_info.rb]: This class is basically a semver parser right? Does it make sense to name it accordingly? What about: `class SemVerParser`?

Comment (jvanbaarsen) [inline lib/version_info.rb]: @joshuajansen Hm, Maybe `VersionParser`?

File: test/jobs/update_server_job_test.rb (hunk 3 above)
Comment (jvanbaarsen) [inline test/jobs/update_server_job_test.rb]: I should have seen that one xD

**Pull request merged**